File: src/main/java/de/tectoast/emolga/utils/sql/base/Condition.java
package de.tectoast.emolga.utils.sql.base;

@SuppressWarnings("unused")
public class Condition {

    public static String and(String con1, String con2) {
        return and(con1, con2, true);
    }

    public static String and(String con1, String con2, boolean p2) {
        return "(" + con1 + (p2 ? " AND " + con2 : "") + ")";
    }

    public static String or(String con1, String con2) {
        return or(con1, con2, true);
    }

    public static String or(String con1, String con2, boolean p2) {
        return p2 ? "(" + con1 + " OR " + con2 + ")" : con1;
    }
}

File: src/main/kotlin/de/tectoast/emolga/utils/sql/base/Condition.kt
package de.tectoast.emolga.utils.sql.base

object Condition {
    fun and(con1: String, con2: String, p2: Boolean = true): String {
        return "(" + con1 + (if (p2) " AND $con2" else "") + ")"
    }

    fun or(con1: String, con2: String, p2: Boolean = true): String {
        return if (p2) "($con1 OR $con2)" else con1
    }
}